librad/src/git/trailer.rs

```diff
@@ -0,0 +1,273 @@
+    message
+        .rsplit("\n\n")
+        .next()
+        .map_or(Ok(None), |trailers_paragraph| {
+            trailers_paragraph
+                .lines()
+                .map(|line| line.trim())
+                .filter(|line| !line.is_empty() && !line.starts_with('#'))
+                .map(|raw_trailer| Trailer::parse(raw_trailer, separators))
+                .collect::<Result<Vec<_>, _>>()
+                .map(Some)
+        })
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_parse_message_with_valid_trailers() {
+        let separator = ':';
+        let msg = r#"
+            A subject that succinctly states what this commit does
+
+            A longer, more detailed paragraph that describes the what's
+            and why's of the changes being introduced by this commit. It
+            is usually capped at 72 chars-long lines.
+
+            # Igored: true
+            Co-authored-by: John Doe <john.doe@test.com>
+            Ticket: #42
+            Tested-by: Tester <tester@test.com>
+            "#;
+        assert_eq!(
+            parse(msg, &[separator]),
+            Ok(Some(vec![
+                Trailer::new("Co-authored-by", "John Doe <john.doe@test.com>").expect(""),
+                Trailer::new("Ticket", "#42").expect(""),
+                Trailer::new("Tested-by", "Tester <tester@test.com>").expect(""),
+            ]))
+        )
+    }
+
+    #[test]
+    fn test_parse_message_trailers_with_supported_separators() {
+        let separators = &[':', '=', '$'];
+        let msg = r#"
+            A subject that succinctly states what this commit does
+
+            A longer, more detailed paragraph that describes the what's
+            and why's of the changes being introduced by this commit. It
+            is usually capped at 72 chars-long lines.
+
+            # Igored: true
+            Co-authored-by: John Doe <john.doe@test.com>
+            Ticket = #42
+            Tested-by $ Tester <tester@test.com>
+            "#;
+        assert_eq!(
+            parse(msg, separators),
+            Ok(Some(vec![
+                Trailer::new("Co-authored-by", "John Doe <john.doe@test.com>").expect(""),
+                Trailer::new("Ticket", "#42").expect(""),
+                Trailer::new("Tested-by", "Tester <tester@test.com>").expect(""),
+            ]))
+        )
+    }
+
+    #[test]
+    fn test_parse_message_with_illformed_trailers() {
+        let separator = ':';
+        let msg = r#"
+            A subject that succinctly states what this commit does
+
+            A longer, more detailed paragraph that describes the what's
+            and why's of the changes being introduced by this commit. It
+            is usually capped at 72 chars-long lines.
+
+            # Ill-formed trailer, missing the token and the separator
+            John Doe <john.doe@test.com>
+            "#;
+        assert!(parse(msg, &[separator]).is_err())
```

Comment:
Is it possible to `assert_eq` on the error we expect? I always find this more useful for 1) documentation purposes 2) debugging purposes. But sometimes it's a PITA, so if that's the case you can ignore this :)

Reply:
Yeah, we should do that, I agree with those criteria. Will do!